### DISCUSSION TITLE
Unable to apply a suggestion to a file conflict to the `main` branch: `This diff has recently been updated.`

### DISCUSSION THREAD

**Discussion by ShamrockLee** [community/community#43052] [Pull Requests]
### Select Topic Area

Bug

### Body

Update: See the [comment](https://github.com/community/community/discussions/43052#discussioncomment-4565795) below for relevent description of the bug.

Apply review suggestions in batch is a great feature!

However, when trying to apply four independent review suggestions in batch, the GitHub web page complains

```
This diff has recently been updated. Please refresh and try again.
```

and fails.

The same problem repeats after doing  the same thing again, and I see no "changes" happen in the mean time.

![Screenshot 2023-01-02 at 00-14-46 singularity fix defaultPath and reflect upstream changes by ShamrockLee · Pull Request #158486 · NixOS_nixpkgs](https://user-images.githubusercontent.com/44064051/210177574-6b44a6d8-622a-4929-9d82-5c0fcbf2a642.png)

**Comment by ShamrockLee** [comment]
Update:

It's not about the batch suggestion commit, but about applying commit suggestions to files that has merge conflict to the branch this PR is to be merged into.

This is still a bug, as the commit-suggestion feature should only concern about the branch that holds the changes in this PR, not the branch to merge the changes into.

![Screenshot 2023-01-02 at 00-33-07 singularity fix defaultPath and reflect upstream changes by ShamrockLee · Pull Request #158486 · NixOS_nixpkgs](https://user-images.githubusercontent.com/44064051/210178067-2ed97a13-3967-44f6-b871-f271205ceb66.png)
![Screenshot 2023-01-02 at 00-34-36 singularity fix defaultPath and reflect upstream changes by ShamrockLee · Pull Request #158486 · NixOS_nixpkgs](https://user-images.githubusercontent.com/44064051/210178069-e51ce5e7-ddf4-4c9e-b45f-a8a53e63f8f6.png)
![Screenshot 2023-01-02 at 00-35-01 singularity fix defaultPath and reflect upstream changes by ShamrockLee · Pull Request #158486 · NixOS_nixpkgs](https://user-images.githubusercontent.com/44064051/210178073-bcac6ddb-f358-4a53-ab64-1064c2a4d36f.png)

**Comment by Frigyes06** [comment]
Can reproduce, super annoying

**Comment by arishojaswi** [comment]
Facing the same issue. Extremely frustrating.

**Comment by isdanni** [comment]
Facing the same issue, wondering when this will be fixed

**Comment by jfullstackdev** [comment]
not sure what situation you are in, but Copilot answered because I asked him too : 

The error message "This diff has recently been updated. Please refresh and try again." typically appears in the context of using GitHub, particularly when you're reviewing pull requests.

This error usually occurs when the pull request you're reviewing has been updated while you were in the process of reviewing it. This could be due to new commits being pushed to the branch, changes being made to the base branch, or the pull request being merged or closed.

The error is essentially telling you that the state of the pull request has changed since you last loaded the page, and you need to refresh the page to see the most recent changes.

To resolve this error, you can simply refresh the page (F5 or Ctrl+R in most browsers) and continue your review with the updated changes. If the error persists, it might be a temporary issue with GitHub, and you could try again after a few minutes.

**Reply by ShamrockLee** [reply]
Please take a look at the [update](https://github.com/community/community/discussions/43052#discussioncomment-4565795). The issue is not about diff view awaiting updating, but about failure to apply suggestions as patches to the feature branch, when the feature branch cannot be merged automatically into the base branch.

This application failure is unexpected, because the suggested changes go to the feature branch instead of the base branch.

**Reply by ShamrockLee** [reply]
Sorry for causing the confusion. I just added an update link on top of the *Body* section.

**Comment by jan-neduchal** [comment]
Still not fixed

**Comment by Andrew-T-Smith** [comment]
Still not fixed as of today. The explanation at the least needs to be improved because I just assumed github was having a service outage.

**Comment by iBug** [comment]
This is now even worse. The UI only shows `There was an error trying to commit changes` and only in F12 can I see the actual error and find this thread.

> <img width="548" height="147" alt="image" src="https://github.com/user-attachments/assets/62898c8b-b85c-4e8b-b7e8-8774b79506f1" />
